app/controllers/admin/recipes_controller.rb

```diff
@@ -0,0 +1,44 @@
+class Admin::RecipesController < ApplicationController
+  http_basic_authenticate_with name: "admin", password: "1234"
+
+  def index
+    @recipes = Recipe.all
+  end
+
+  def show
+    @recipe = Recipe.find(params[:id])
+  end
+
+  def new
+    @recipe = Recipe.new
+  end
+
+  def create
+    @recipe = Recipe.new(recipe_params)
+
+    @recipe.save
+    redirect_to @recipe
+  end
+
+  def update
+    @recipe = Recipe.find(params[:id])
+
+    if @recipe.update(params[:id])
+      redirect_to @recipe
+    else
+      render 'edit' #this is a connection to the view
+    end
+  end
+
+  def destroy
+    @recipe = Recipe.find(params[:id])
+    @recipe.destroy
+
+    redirect_to admin/recipe_path
```

Comment:
as a string "admin/recipe_path" or better admin_recipe_path (check correct one in `rake routes`)